internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -520,8 +538,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
 	}
 
-	err := r.client.Apply(ctx, cp, resource.AllowUpdateIf(func(old, obj runtime.Object) bool { return !cmp.Equal(old, obj) }))
+	var err error
+	if meta.WasCreated(cp) {
+		err = r.client.Apply(ctx, cp, resource.AllowUpdateIf(func(old, obj runtime.Object) bool { return !cmp.Equal(old, obj) }))
+	} else {
+		// if composite did not exist at the beginning of the loop, we want to create it
+		// so that we can check if the composite name is not already taken
+		err = r.client.Create(ctx, cp)
+	}
 	switch {
+	case kerrors.IsAlreadyExists(err):
+		// generated name is already taken
+		// let's requeue and try again with the new name
+		log.Debug("Cannot create composite, another already exists with that name. Requeue and try another name")
+		err = errors.Wrap(err, errCreateComposite)
+		record.Event(cm, event.Warning(reasonCompositeConfigure, err))
+		cm.SetConditions(xpv1.ReconcileError(err))
```

Comment:
would stay silent. No event, no conditions. This is part of an to be expected conflict. Nothing to report to the user.